### PR TITLE
feaLib: without glyphNames, warn but don't parse hyphens as ranges

### PR DESCRIPTION
**Pull request by anthrotype**
Fixes https://github.com/fonttools/fonttools/issues/1768

cc @lianghai

### CODE REVIEW

**Comment by anthrotype** [comment]
If Parser is initialized without a glyphNames parameter (or with empty one), since it cannot distinguish between a glyph name containing an hyphen, or a range of glyph names, it will now interpret them as literal glyph names, while also outputting a logging warning to alert user about the ambiguity.
To silence the warning, one should either provide an explicit glyphNames to the Parser, or wrap the range-denoting hyphen with spaces, so that it cannot be confused with a single glyph name.

File: Lib/fontTools/feaLib/parser.py [hunk 1]
```diff
@@ -350,8 +355,8 @@ def parse_glyphclass_(self, accept_glyphname):
                 glyphs.add_class(gc)
             else:
                 raise FeatureLibError(
-                    "Expected glyph name, glyph range, "
-                    "or glyph class reference",
+                    f"Expected glyph name, glyph range, "
```
Comment (madig) [inline Lib/fontTools/feaLib/parser.py:358]: `f` not necessary for upper string.

File: Lib/fontTools/feaLib/parser.py [hunk 2]
```diff
@@ -314,6 +314,11 @@ def parse_glyphclass_(self, accept_glyphname):
                         start, limit,
                         self.make_glyph_range_(location, start, limit))
                 else:
+                    if not self.glyphNames_:
+                        log.warning(str(FeatureLibError(
+                            f"Ambiguous glyph name that looks like a range: {glyph!r}",
```
[the remark this answers is not in the document]
Comment (anthrotype) [reply Lib/fontTools/feaLib/parser.py:319]: Ouch! Thanks for spotting 😅

**Comment by anthrotype** [comment]
What do we want to do here? I think it's perfectly reasonable to expect that, when the parser is not given a context to distinguish glyph names containing hyphens from range of glyph names (e.g. when the content of the glyph set is not known in advance, and the purpose of such parsing is not building binary tables but extract/manipulate the data in the FEA), that the parser treats any token that resembles a glyph name as a glyph name while warning that it may resemble a glyph range. The warning gives the user the possibility to modify the text so that the ambiguity is resolved (by inserting spaces around the range-denoting hyphens) or by providing an explicit glyphSet when initializing the Parser.
This will not change the way feaLib builds the OTL tables since we always require that the glyphSet is given when the Builder instantiate its Parser, thus it is not an incompatible change.

I am inclined to merge this, but I want to hear from others first.

**Pull request merged**